stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Final, Literal, TypeVar, final
+from typing_extensions import Buffer, Self
+
+_T = TypeVar("_T")
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+# Mark as final, since instantiating ChannelID is not supported.
+@final
+class ChannelID:
+    @property
+    def end(self) -> Literal["send", "recv", "both"]: ...
+    @property
+    def send(self) -> Self: ...
+    @property
+    def recv(self) -> Self: ...
+    def __eq__(self, other: Self) -> bool: ...
+    def __ge__(self, other: Self) -> bool: ...
+    def __gt__(self, other: Self) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __index__(self) -> int: ...
+    def __int__(self) -> int: ...
+    def __le__(self, other: Self) -> bool: ...
+    def __lt__(self, other: Self) -> bool: ...
+    def __ne__(self, other: Self) -> bool: ...
```

Comment:
`__eq__` and `__ne__` are not like the other comparison dunders: they nearly always take `object` for their second argument, which is why mypy is now complaining. It's not always possible to use `>` to compare two arbitrary Python objects, but it's nearly always possible to use `==` or `!=` to compare two arbitrary Python objects:

```suggestion
    def __eq__(self, other: object) -> bool: ...
    def __ge__(self, other: Self) -> bool: ...
    def __gt__(self, other: Self) -> bool: ...
    def __hash__(self) -> int: ...
    def __index__(self) -> int: ...
    def __int__(self) -> int: ...
    def __le__(self, other: Self) -> bool: ...
    def __lt__(self, other: Self) -> bool: ...
    def __ne__(self, other: object) -> bool: ...
```

Reply:
Interesting, didn't realize! Fixed: https://github.com/python/typeshed/pull/12199/commits/d45c6f72e0b4effe9d84b07f59ef27b4c39d9966